tests/test_using.py

```diff
@@ -1137,6 +1137,23 @@ class Meta:
         obj = TestObjectFactory.build()
         self.assertEqual(42, obj.attributes)
 
+    def test_rename_non_existent_kwarg(self):
+        '''
+        If a key is specified in the rename attribute of the model factory, but
+        not passed into the object when instantiated we should bypass the
+        keyword re-assignment instead of throwing an error.
+        https://github.com/FactoryBoy/factory_boy/issues/504
+        '''
+        class TestObject(object):
+            def __init__(self, attributes=None):
+                self.attributes = attributes
+
+        class TestObjectFactory(factory.Factory):
+            class Meta:
+                model = TestObject
+                rename = {'attributes_': 'attributes'}
```

Comment:
Perhaps we could use the same models as in the documentation, to avoid this surprising variable name?

```python
             class ImageFactory(factory.Factory):
                 # The model expects "attributes"
                 form_attributes = ['thumbnail', 'black-and-white']
 
                 class Meta:
                     model = Image
                    rename = {'form_attributes': 'attributes'}
```

Reply:
Sorry, not fully catching the intent here. Do you mean that I should replace the class name `TestObjectFactory` with `ImageFactory` and then use a mis-spelled version of `form_attributes` for the test? For reference, I was trying to stay consistent with the name schemes of the existing tests of the `rename` functionality.

https://github.com/FactoryBoy/factory_boy/blob/master/tests/test_using.py#L1126

Reply:
I did not realize the factory was reused from `test_rename` above, sorry about that.

My recommendation is to use a factory that's more realistic. `attributes_` is an odd name for an attribute. I suggest using the factory from the documentation (https://factoryboy.readthedocs.io/en/latest/reference.html#factory.FactoryOptions.rename).

What do you think about:

1. Making a first commit to replace the `TestObject` with the `ImageFactory` definition from the documentation,
2. Move the object and factory definition outside of the test in order for both `test_rename` and `test_rename_non_existent_kwarg` to use it?

---

If replacing `TestObjectFactory` with an `ImageFactory` is not appealing to you or represent too much work, I don't mind handling that in a separate PR.